content/beginner/194_secrets_manager/prepare-secret.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Prepare secret and IAM access controls"
+date: 2021-10-01T00:00:00-04:00
+weight: 20
+pre: '<i class="fa fa-film" aria-hidden="true"></i>'
+draft: false
+---
+
+#### Set variables
+Verify AWS_REGION variable. 
+```bash
+test -n "$AWS_REGION" && echo AWS_REGION is "$AWS_REGION" || echo AWS_REGION is not set
+```
+If not set, then complete [Start the workshop instructions](/020_prerequisites/workspaceiam/) and verify again.
+
+Set the cluster name variable.
+```bash
+export EKS_CLUSTERNAME="eksworkshop-eksctl"
+```
+
+#### Create a test secret with the AWS Secrets Manager.
+```bash
+aws --region "$AWS_REGION" secretsmanager \
+  create-secret --name DBSecret_eksworkshop \
+  --secret-string '{"username":"foo", "password":"super-sekret"}'
+
+```
+
+Get secret's ARN.
+
+```bash
+SECRET_ARN=$(aws --region "$AWS_REGION" secretsmanager \
+    describe-secret --secret-id  DBSecret_eksworkshop \
+    --query 'ARN' | sed -e 's/"//g' )
+
+echo $SECRET_ARN
+```
+
+You can also verify the newly created secret via AWS console under [AWS Secrets Manager](https://console.aws.amazon.com/secretsmanager/home?#!/secret?name=DBSecret_eksworkshop) 
+
+![DBSecret_eksworkshop](/images/secret-in-secretsManager.png)
+
+
+#### Create an IAM policy
+Create an IAM with permissions to access the secret.
+
+```bash
+IAM_POLICY_NAME_SECRET="DBSecret_eksworkshop_secrets_policy_$RANDOM"
+IAM_POLICY_ARN_SECRET=$(aws --region "$AWS_REGION" iam \
+	create-policy --query Policy.Arn \
+    --output text --policy-name $IAM_POLICY_NAME_SECRET \
+    --policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [ {
+        "Effect": "Allow",
+        "Action": ["secretsmanager:GetSecretValue", "secretsmanager:DescribeSecret"],
+        "Resource": ["'"$SECRET_ARN"'" ]
+    } ]
+}')
+
+echo $IAM_POLICY_ARN_SECRET | tee -a 00_iam_policy_arn_dbsecret
```

Comment:
Why is this written to a file as well?

Reply:
For a better cleanup experience later.

 Value of $IAM_POLICY_ARN_SECRET written to the file 00_iam_policy_arn_dbsecret, so it is easier to clean up this secret later using a stored value in the file. Otherwise, if the user runs cleanup steps later in a different shell window, then variable $IAM_POLICY_ARN_SECRET may not be present there and errors out.